rust/src/measurements/laplace/float/make_vector_float_laplace_cks20.tex

```diff
@@ -0,0 +1,116 @@
+\documentclass{article}
+\input{../../../lib.sty}
+
+
+\title{\texttt{fn make\_vector\_float\_laplace\_cks20}}
+\author{Michael Shoemate}
+\begin{document}
+
+\maketitle
+
+\contrib
+
+Proves soundness of \rustdoc{measurements/fn}{make\_vector\_float\_laplace\_cks20} in \asOfCommit{mod.rs}{f5bb719}.
+The function on the resulting measurement takes in a data set \texttt{x} (a vector of floats), 
+and returns a sample from the Vector Discrete Laplace Distribution centered at \texttt{x}, with a fixed noise scale.
+
+\subsection*{PR History}
+\begin{itemize}
+    \item \vettingPR{490}
+\end{itemize}
+
+\section{Hoare Triple}
+
+\subsection*{Preconditions}
+\begin{itemize}
+    \item Variable \texttt{input\_domain}, of type \texttt{VectorDomain<AtomDomain<T>>}
+    \item Variable \texttt{input\_metric}, of type \texttt{L1Distance<T>}
+    \item Variable \texttt{scale}, of type \texttt{QO}
+    \item Variable \texttt{k}, of type \texttt{Option<i32>}
+    \item Type \texttt{T} must have trait \rustdoc{traits/trait}{Float} and \rustdoc{traits/trait}{CastInternalRational}
+    \item Type \texttt{i32} must be constructable from the bit representation of \texttt{T} (used to calibrate relaxation)
+\end{itemize}
+
+\subsection*{Pseudocode}
+\lstinputlisting[language=Python,firstline=2,escapechar=|]{./pseudocode/make_vector_float_laplace_cks20.py}
+
+\subsection*{Postcondition}
+
+\validMeasurement{\texttt{(input\_domain, input\_metric, scale, k, T)}}{\\\texttt{make\_vector\_float\_laplace\_cks20}}
+
+\section{Proof}
+
+\begin{proof} 
+\textbf{(Privacy guarantee.)} 
+    
+\begin{tcolorbox}
+    The proof assumes the following lemma.
+    \begin{lemma}
+        \texttt{get\_discretization\_consts}, \texttt{sample\_discrete\_laplace\_Z2k} and \texttt{laplace\_map} each satisfy their postcondition.
+    \end{lemma}
+\end{tcolorbox}
+
+This mechanism can be thought of as a stable transformation from a vector of floating-point numbers to a vector of rationals,
+followed by a mechanism that adds noise to each rational number.
+
+\subsection{Rounding Transformation}
+
+The transformation rounds the input to the nearest rational number where the denominator is no greater than $2^k$.
+By the postcondition of \texttt{get\_discretization\_consts}, this rounding changes each value by at most \texttt{relaxation}.
+
+\begin{align*}
+    &\max_{x \sim x'} d_{L1}(round(x), round(x')) \\
+    =& \max_{x \sim x'} \sum_i^d \left| round(x_i) - round(x'_i) \right| \\
```

Comment:
I'm expecting something like `i=0` instead of just `i` under the sigma, but maybe this is not an usual convention?

Reply:
You're right, it would be less ambiguous to specify it.